src/main/host/descriptor/socket.c

```diff
@@ -84,7 +84,7 @@ static void _legacysocket_close(LegacyDescriptor* descriptor, Host* host) {
 
     Tracker* tracker = host_getTracker(host);
     if (tracker != NULL) {
-        tracker_removeSocket(tracker, descriptor_getHandle(descriptor));
+        tracker_removeSocket(tracker, (guintptr)descriptor);
```

Comment:
This commit introduces a lot of casting to `guintptr`, which seems error-prone. It'd be nice to encapsulate at least the cast itself. e.g. could these various `tracker_*` APIs take a `Descriptor*` or `Socket*`, and maybe in the tracker have a private helper for converting to a `guintptr` "handle"?

e.g.:

```
static guintptr _tracker_socketHandle(Descriptor* d) { 
  // XXX:  utility_assert to check that the Descriptor* is a Socket?
  return (guintptr) d;
}

void tracker_removeSocket(Tracker* tracker, Descriptor* d) {
  guintptr handle = _tracker_socketHandle(d);
  ...
}
```

Maybe even better to use a specialized value type instead of a guintptr, but maybe that's overkill if it can be purely internal to the Tracker.

Reply:
My reason for using a `guintptr` was so that the `Tracker` stays independent of the socket type (for example the rust `Socket` or the C `LegacySocket` could both be used), and so that the `Tracker` doesn't try to call socket methods. But I don't think these are a big deal, so I'll change it to take pointers and cast them to `guintptr` internally. We'd probably want to rewrite this in rust anyways before calling it from rust sockets.